contracts/src/systems/order/take_fungible_order.cairo

```diff
@@ -232,361 +233,301 @@ mod TakeFungibleOrder {
+    }
+
+
+
+
+
+    #[test]
+    #[available_gas(30000000000000)]
+    fn test_take_trade_without_caravan() {
+
+        let (
+            world,
+            maker_id,
+            taker_id, 
+            trade_id, 
+            maker_order_id, 
+            taker_order_id
+        ) = setup(false);
+
+
+        // taker takes trade
+        starknet::testing::set_contract_address(
+            contract_address_const::<'taker'>()
+        );
+        let mut calldata = array![];
+        Serde::serialize(@taker_id, ref calldata);
+        Serde::serialize(@trade_id, ref calldata);
+        world.execute('TakeFungibleOrder', calldata);
+
+
+
+        // taker wood balance should be 500
+        let taker_wood_resource = get!(world, (taker_id, ResourceTypes::WOOD), Resource);
+        assert(taker_wood_resource.balance == 500, 'resource balance should be 500'); // 600 - 100
+
+        // taker stone balance should be 400
+        let taker_stone_resource = get!(world, (taker_id, ResourceTypes::STONE), Resource);
+        assert(taker_stone_resource.balance == 400, 'resource balance should be 400'); // 600 - 200
+
+
+        // status should be accepted
+        let status = get!(world, trade_id, Status);
+        assert(status.value == 1, 'status should be accepted');
+
+
+        // verify arrival time of maker order
+        let maker_arrival_time = get!(world, maker_order_id, ArrivalTime);
+        assert(maker_arrival_time.arrives_at == 0, 'arrival time should be 0');
+ 
+
+        // verify arrival time of taker order
+        let taker_arrival_time = get!(world, taker_order_id, ArrivalTime);
+        assert(taker_arrival_time.arrives_at == 0, 'arrival time should be 0');
+       
+
+        // verify position of maker order
+        let maker_order_position = get!(world, maker_order_id, Position);
+        assert(maker_order_position.x == 45, 'position x should be 45');
+        assert(maker_order_position.y == 50, 'position y should be 50');
+
+
+        // verify position of taker order
+        let taker_order_position = get!(world, taker_order_id, Position);
+        assert(taker_order_position.x == 60, 'position x should be 60');
+        assert(taker_order_position.y == 70, 'position y should be 70');
+    }
+
+
+
+
+
+
+    #[test]
+    #[available_gas(30000000000000)]
+    fn test_take_trade_with_caravan() {
+        
+        let (
+            world,
+            maker_id,
+            taker_id, 
+            trade_id, 
+            maker_order_id, 
+            taker_order_id
+        ) = setup(true);
+
+        // create a caravan owned by the maker
+        let maker_caravan_id = 20_u64;
+        let maker_caravan_id_felt: felt252 = maker_caravan_id.into();
+
+        set!(world, (Owner { address: contract_address_const::<'maker'>(), entity_id: maker_caravan_id.into()}));
+        set!(world, (Position { x: 100_000, y: 200_000, entity_id: maker_caravan_id.into()}));
```

Comment:
you should set the position of the maker caravan on the maker position, because that is the only way to attach the caravan.
Same for the taker caravan position which should be on the same position as the taker.

Reply:
Got it. Fixed. Updated the taker and maker positions to large values also because the distance between small values like  `Position {45, 60}` and `Position {60,70}` would be 0, making travel time 0

Reply:
ok